src/components/shop-app.js

```diff
@@ -294,169 +308,47 @@ class ShopApp extends connect(store)(LitElement) {
     loadComplete: Boolean
   }}
 
-  _propertiesChanged(props, changed, oldProps) {
-    if (changed) {
-      if ('categoryName' in changed) {
-        this._categoryNameChanged(props.categoryName, oldProps.categoryName);
-      }
-      if ('page' in changed) {
-        this._pageChanged(props.page, oldProps.page);
-      }
-      if ('offline' in changed) {
-        this._offlineChanged(props.offline, oldProps.offline);
-      }
-      if ('meta' in changed) {
-        this._metaChanged(props.meta, oldProps.meta);
-      }
+  didRender(props, changed, oldProps) {
+    if ('page' in changed || 'categoryName' in changed) {
+      // TODO: For list view, scroll to the last saved position only if the category has not changed
+      scroll({ top: 0, behavior: 'silent' });
+    }
+    if ('page' in changed) {
+      // TODO: Remove this when app-header updated to use ResizeObserver so we can avoid this bit.
+      // The size of the header depends on the page (e.g. on some pages the tabs
+      // do not appear), so reset the header's layout when switching pages.
+      const header = this.shadowRoot.querySelector('#header');
+      header.resetLayout();
     }
-    super._propertiesChanged(props, changed, oldProps);
   }
 
   constructor() {
     super();
 
-    installRouter(() => this._updateLocation());
-    installNetwork(store);
     store.dispatch(fetchCategories());
+    installRouter(() => this._updateLocation());
+    installOfflineWatcher((offline) => this._offlineChanged(offline));
+    installMediaQueryWatcher('(max-width: 767px)', (matches) => this._smallScreen = matches);
   }
 
   stateChanged() {
     const state = store.getState();
```

Comment:
Note to self: `stateChanged(state)` is passed the state, so `getState()` isn't needed.